### PR TITLE
Connect backend to frontend

### PR DESCRIPTION
**Pull request by Lev-Stambler**
- Change the dummy class to simulate and return results
- Add frontend to display results

### CODE REVIEW

**Comment by Lev-Stambler** [comment]
Screenshot_2020-07-20_070903

File: biogrid/libs/biogrid-simulator/src/lib/biogrid-state/biogrid-state-class.ts [hunk 1]
```diff
@@ -15,24 +15,25 @@ export class BiogridState implements StateGraph {
...
-      }
-    }
-    this.graph.setNode(grid.name, (grid as GridItem));
+        return { x: 0, y: 0 };
+      },
+    };
+    this.graph.setNode(grid.name, grid as GridItem);
 
     // Add all the vertices as nodes/vertices of the graph, with a name for
     // the particular grid item and label which is data for the particular vertex as the GridItem itself
```
Comment (aosmann7) [inline biogrid/libs/biogrid-simulator/src/lib/biogrid-state/biogrid-state-class.ts]: Capitalize comment

File: biogrid/libs/biogrid-simulator/src/lib/biogrid-state/biogrid-state-class.ts [hunk 2]
```diff
@@ -126,29 +136,49 @@ export class BiogridState implements StateGraph {
...
     const newVertexName = newVertex.name;
     for (const vertex of this.graph.nodes()) {
-      const distance = this.calculateDistance(newVertex, (this.graph.node(vertex)) as StateGraphVertex)
+      const distance = this.calculateDistance(
+        newVertex,
+        this.graph.node(vertex) as StateGraphVertex
+      );
       let edge: StateGraphEdge;
       // Solar panels to the grid only
       // searching for includes GRID so that when scaling it is easy to add multiple grids
```
Comment (aosmann7) [inline biogrid/libs/biogrid-simulator/src/lib/biogrid-state/biogrid-state-class.ts]: capitalize here

File: biogrid/apps/biogrid-mvp/src/app/pages/simulate-page/simulate-page.tsx [hunk 3]
```diff
@@ -34,8 +36,20 @@ export const SimulatePage = () => {
               <td>Energy wasted in transport</td>
               <td>{simulationResults.energyWastedInTransportation}</td>
             </tr>
-            
+            {/* <ol> */}
```
Comment (aosmann7) [inline biogrid/apps/biogrid-mvp/src/app/pages/simulate-page/simulate-page.tsx]: What is this for?

**Review by aosmann7 — APPROVED** [review]

File: biogrid/libs/grid-simulator/src/lib/state/state-interface.ts [hunk 4]
```diff
@@ -20,6 +20,7 @@ export interface StateGraph {
   getAllPositions: () => ItemPosition[];
   getGridItem: (name: string) => GridItem;
   getGraph: () => Graph;
+  getJsonGraph: () => any;
```
Comment (RolandNaijuka) [inline biogrid/libs/grid-simulator/src/lib/state/state-interface.ts:23]: this is not implemented anywhere? could you remove it or is it implemented?

File: biogrid/apps/api/src/app/services/Biogrid.ts [hunk 5]
```diff
@@ -11,21 +18,46 @@ export interface NewBiogridOpts {
...
+    new Building(10, 1, 2),
+    new Building(10, 3, 1),
+  ];
+  const town = new RuralArea(buildings, 10, 10);
+  biogrid = new Biogrid(town, {
+    numberOfLargeBatteryCells: body.largeBatteryCells,
+    numberOfSmallBatteryCells: body.smallBatteryCells,
+    numberOfSolarPanels: 10,
+  });
+  return 'Created';
```
Comment (RolandNaijuka) [inline biogrid/apps/api/src/app/services/Biogrid.ts:43]: are there cases when the Biogrid is not created? if so, what is returned? If not, do you have to return this?

File: biogrid/apps/api/src/app/services/Biogrid.ts [hunk 6]
```diff
@@ -11,21 +18,46 @@ export interface NewBiogridOpts {
...
 
+// TODO change to a stateless solution
+// See issue: https://github.com/googleinterns/step141-2020/issues/50
+let biogrid: Biogrid;
+const biobrain = BioBrain.Instance;
+const states: any[] = [];
+
+// TODO, allow users to specify number of buildings, town size, and number of Solar Panels
+// See issue: https://github.com/googleinterns/step141-2020/issues/49
 export async function createNewBiogrid(body: NewBiogridOpts) {
```
Comment (RolandNaijuka) [inline biogrid/apps/api/src/app/services/Biogrid.ts]: async should return a promise

File: biogrid/apps/api/src/app/services/Biogrid.ts [hunk 7]
```diff
@@ -11,21 +18,46 @@ export interface NewBiogridOpts {
...
+  const town = new RuralArea(buildings, 10, 10);
+  biogrid = new Biogrid(town, {
+    numberOfLargeBatteryCells: body.largeBatteryCells,
+    numberOfSmallBatteryCells: body.smallBatteryCells,
+    numberOfSolarPanels: 10,
+  });
+  return 'Created';
 }
 
 export async function runBiogridSimulation() {
```
Comment (RolandNaijuka) [inline biogrid/apps/api/src/app/services/Biogrid.ts]: async should return a promise

**Review by RolandNaijuka — CHANGES_REQUESTED** [review]

File: biogrid/libs/grid-simulator/src/lib/state/state-interface.ts (hunk 4 above)
Comment (Lev-Stambler) [reply biogrid/libs/grid-simulator/src/lib/state/state-interface.ts:23]: It is in biogrid state and biogrid

File: biogrid/apps/api/src/app/services/Biogrid.ts (hunk 5 above)
Comment (Lev-Stambler) [reply biogrid/apps/api/src/app/services/Biogrid.ts:43]: Yeah, it should throw if it is not created. Error handling is automatically handled by Tsoa then

**Review by RolandNaijuka — APPROVED** [review]

**Pull request merged**